### PR TITLE
Revert fix/slug-to-classname

### LINKED ISSUE #32347: Different class names for preset slugs
Themes provide some presets to the editor, such as font sizes and colors, either via the `add_theme_support` https://github.com/WordPress/gutenberg/issues/26940 or the `theme.json` https://github.com/WordPress/gutenberg/issues/31629

The slug of the preset values is used to generate a class name in two different situations:

1. when in the editor, to generate the class name to be attached to the block markup upon an user action, as in:

```html
<p class="has-slug-preset">Content</p>
```

2. to generate the classes to be enqueued as part of the global styles stylesheet:

```css
.has-slug-preset { ... }
```

## What's the issue

When the slug contains a number, the class attached to the block markup and the class generated for the stylesheet are different. See issues reported at https://github.com/WordPress/gutenberg/issues/31629 and https://github.com/WordPress/gutenberg/issues/26940 There's also issues with slugs that contain camelCased words.

Slug | Class generated in the editor | Class generated in the front
--- | --- | --- 
`slug1` | `.has-slug-1-preset` | `.has-slug1-preset`
`2slug` | `.has-2-slug-preset` | `.has-2slug-preset`
`slug3number` | `.has-slug-3-number-preset` | `.has-slug3number-preset`
`4th` | `.has-4th-preset` | `.has-4th-preset`
`5sl` | `.has-5-sl-preset` | `.has-5sl-preset`
`slugNumber` | `.has-slug-number-preset` | `.has-slugnumber-preset`

Notice how the server doesn't process the slug but the client splits by number, except if the slug is an English ordinal such as `4th`. The client also converts camelCased slugs into kebab-cased slugs, while the server doesn't.

## Why this happens

In the client, both for font-sizes and colors we use the lodash's [kebabCase](https://lodash.com/docs/4.17.15#kebabCase) function to process the incoming slug such as:

```js
has-${ kebabCase( slug ) }-preset
```

The lodash's kebabCase function has the particularity that it considers numbers as something to separate as well, [unless they're English ordinals](https://github.com/lodash/lodash/issues/2843#issuecomment-262255270).

In the server, we don't process the slug but output it as it is:

```php
'.has-' . $value['slug'] . '-' . $class['class_suffix']
```

## How to fix it

1. Do not do any processing to the slug in the client, but use the incoming value. Existing classes would still work: the old slug is recognized as an extra class and because the theme should have enqueued it, it'll still work.

2. Replicate the lodash's kebabCase logic (including English ordinals) in the server.

3. Use our own regexp / function to do the same in both places, to make sure changes to lodash's kebabCase don't affect us.

### PR DESCRIPTION
**Pull request by oandregal**
This reverts https://github.com/WordPress/gutenberg/pull/32352 that by trying to fix https://github.com/WordPress/gutenberg/issues/32347 created issues https://github.com/WordPress/gutenberg/issues/32719 and https://github.com/WordPress/gutenberg/issues/32721

Third parties (such as Jetpack) rely on specific behaviours from lodash's `kebabCase` method (see the table at https://github.com/WordPress/gutenberg/issues/32347). Because the generated classes are attached to the markup we can't really remove kebabCase as that's the source of the behaviour in use. It seems the way to fix this is by implementing the same behavior around numbers and numerals in the server.

While we do that we can gain some time by reverting https://github.com/WordPress/gutenberg/pull/32352 

## How to test

― Test 1

- Install and activate Jetpack on a site. This requires a public site. You can created one at jurassic.ninja for example.
- Download the gutenberg plugin from this PR (see artifacts in the CI tab). 
- Activate the subscription form block with [these instructions](https://jetpack.com/support/jetpack-blocks/subscription-form-block/).
- Insert the following code into code view of editor.

```html
<!-- wp:jetpack/subscriptions {"customButtonBackgroundColor":"#151516","fontSize":16,"customFontSize":16,"borderRadius":3,"borderColor":"#ffffff","customBorderColor":"#ffffff"} -->
<div class="wp-block-jetpack-subscriptions wp-block-jetpack-subscriptions__supports-newline">[jetpack_subscription_form show_subscribers_total="false" button_on_newline="false" custom_background_button_color="#151516" custom_font_size="16" custom_border_radius="3" custom_border_weight="1" custom_border_color="#ffffff" custom_padding="15" custom_spacing="10" submit_button_classes="has-16-font-size has-ffffff-border-color" email_field_classes="has-16-font-size has-ffffff-border-color" show_only_email_and_button="true"]</div>
<!-- /wp:jetpack/subscriptions -->
```
- Update post.
- The block should display as valid.

― Test 2

- Using TT1-blocks update the `green` slug to `green2green` in the `theme.json`.
- Go to the editor and attach that color text to a block. Verify that the generated class is `has-green-2-green-color`.
- Go to the front-end and inspect the contents of the `global-styles-inline-css` style tag. Verify that
  - the generated classes are `has-green2green-color`, `has-green2green-background-color` and `has-green2green-border-color` 
  - the CSS custom property is `--wp--preset--color--green2green`.
- Load the site editor and verify that it behaves like the front end (search for `has-green2green*` classes, etc.

### CODE REVIEW

File: packages/block-editor/src/components/colors/test/utils.js [hunk 1]
```diff
@@ -109,10 +109,10 @@ describe( 'color utils', () => {
 			).toBeUndefined();
 		} );
 
-		it( 'should return a class name with the color slug without spaces', () => {
-			expect(
-				getColorClassName( 'background', 'Light   Purple veryDark' )
-			).toBe( 'has-Light-Purple-veryDark-background' );
```
Comment (ockham) [inline packages/block-editor/src/components/colors/test/utils.js]: How about we retain this? I think it's a more powerful check than the other one.

**Comment by ockham** [comment]
I was thinking to maybe add a few of the unit test cases from #32720 and #32722 (minus the `console.warn` checks) to better describe those helper functions' expected behavior here (for any future implementation changes) 😄

**Comment by ockham** [comment]
> I was thinking to maybe add a few of the unit test cases from #32720 and #32722 (minus the `console.warn` checks) to better describe those helper functions' expected behavior here (for any future implementation changes) 😄

Done ✅

**Comment by oandregal** [comment]
Updated the tests instructions in the PR and manually tested them myself. It works.

**Comment by ockham** [comment]
> ― Test 1

Tested ✅

**Review by ockham — APPROVED** [review]
Looking good! Thanks @nosolosw ❤️

**Comment by ockham** [comment]
> ― Test 2

Tested ✅

**Comment by oandregal** [comment]
I see there's a test failure: "TypeWriter › should maintain caret position after scroll" that has been reported to fail in other PRs as well. I'd like to address it and help fixing that test but also need to merge this to release 10.8.2, so I'm going to go ahead and merge.

**Pull request merged**